pycyphal/application/file.py

```diff
@@ -500,10 +500,280 @@ def __repr__(self) -> str:
         return pycyphal.util.repr_attributes(self, self._node, server_node_id=self._server_node_id)
 
 
+class FileClient2:
+    """
+    A trivial proxy that provides a higher-level and more pythonic API on top of the standard RPC-services
+    from ``uavcan.file``.
+    Client instances are created lazily at first request and then kept alive until this instance is closed.
+    All remote operations raise :class:`FileTimeoutError` on timeout.
+
+    In contrast to pycyphal.application.file.FileClient,
+    pycyphal.application.file.FileClient2 raises exceptions for errors reported
+    over the network. The intent is to expose better error handling in the API;
+    especially avoid ``isintance()`` tests on return values.
```

Comment:
Please use ReST formatting.

```suggestion
    In contrast to :class:`FileClient`,
    :class:`FileClient2` raises exceptions for errors reported
    over the network.
    The intent is to expose more pythonic error handling in the API.
```

Reply:
I'm not familiar with it, but I'll give it a try :)